cmd/promtool/main.go

```diff
@@ -364,6 +384,46 @@ func (ls lintConfig) lintDuplicateRules() bool {
 	return ls.all || ls.duplicateRules
 }
 
+const promDefaultURL = "http://localhost:9090"
+
+// Check server health & readiness.
+func CheckServerHealthAndReadiness(serverURL, check string) error {
+	if serverURL == "" {
+		serverURL = promDefaultURL
+	}
+
+	u, err := url.Parse(serverURL)
+	if err != nil {
+		return fmt.Errorf("error parsing URL: %s", serverURL)
+	}
+
+	if check == checkHealth {
+		u.Path += "/-/healthy"
+	} else if check == checkReadiness {
+		u.Path += "/-/ready"
+	}
```

Comment:
Adding these checks here beat the purpose of having a general utility function. We can skip this whole block of checks by passing these health and readiness values from the caller function.

1. Replace your consts by the actual values:
       ```
        checkHealth              = "/-/healthy"
	checkReadiness       = "/-/ready" 
```
2. Call this function using these consts as arguments and use them in setting u.Path value:
       ```CheckServerHealthAndReadiness(*serverHealthURLArg, checkParameter)```

3. [Suggestion] The name of this function should be general so that it can be used by other functions in future.